### PR TITLE
ESM3 Protein fold example

### PR DESCRIPTION
**Pull request by shariqm-modal**
Fold Proteins using ESM3 and visualize confidence in structure and known structures from PDBs

Screenshot 2024-12-04 at 1 46 47 PM

### Type of Change

- [x] New example
- [ ] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

## Checklist

- [x] Example is testable in synthetic monitoring system, or `lambda-test: false` is added to example frontmatter (`---`)
  - [x] Example is tested by executing with `modal run` or an alternative `cmd` is provided in the example frontmatter (e.g. `cmd: ["modal", "deploy"]`)
  - [x] Example is tested by running with no arguments or the `args` are provided in the example frontmatter (e.g. `args: ["--prompt", "Formula for room temperature superconductor:"]`
- [x] Example is documented with comments throughout, in a [_Literate Programming_](https://en.wikipedia.org/wiki/Literate_programming) style.
- [x] Example does _not_ require third-party dependencies to be installed locally
- [x] Example pins its dependencies
  - [x] Example pins container images to a stable tag, not a dynamic tag like `latest`
  - [x] Example specifies a `python_version` for the base image, if it is used
  - [x] Example pins all dependencies to at least minor version, `~=x.y.z` or `==x.y`
  - [x] Example dependencies with `version < 1` are pinned to patch version, `==0.y.z`

## Outside contributors

You're great! Thanks for your contribution.

### CODE REVIEW

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-2fce074.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-49a1375.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-3b50946.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-ffeec26.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-e0f2cb8.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-be2e81c.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-4cde486.modal.run

**Review by charlesfrye — CHANGES_REQUESTED** [review]

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-3357c03.modal.run

**Comment by shariqm-modal** [comment]
Summary of updates:
- Plugged in molstar and removed `py3Dmol` and my wrapper around it
- Replaced PDB id with UniProt Accession Numbers which is more modern and contains links to more databases
- Use a FASTA database to read sequence data more clean 
- Return mmCIF bytes instead of an ESMProtein for the inference class

Punts (b/c taking too long):
- Not plotting secondary structure of ESM Output
- Not plotting pLDDT confidence of ESM Output
- Not plotting PDB structure of known properties

gradio_ui

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-b258ecd.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-7610d86.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-daf4d60.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-592a10d.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-ee736f3.modal.run

File: 06_gpu_and_ml/protein-folding/esm3.py [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+
+MINUTES = 60  # seconds
+
+app = modal.App("example-esm3-dashboard")
+
+# ### Create a Volume to store cached ESM3 model and Entrez sequence data
+
+# To minimize cold start times we'll store the ESM3 model weights on a Modal
+# [Volume](https://modal.com/docs/guide/volumes). Normally we would do that
+# through the `cache_dir` argument of `from_pretrained` but ESM3 doesn't
```
[the remark this answers is not in the document]
Comment (charlesfrye) [reply 06_gpu_and_ml/protein-folding/esm3.py]: In my updates, I added a link to the model weights guide -- that way we can defer discussion of finer points.

**Review by charlesfrye — COMMENTED** [review]
One note about structure, on top of the changes I made.

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-fc30e47.modal.run

**Review by charlesfrye — APPROVED** [review]

**Pull request merged**